prow/git/v2/adapter.go

```diff
@@ -59,6 +59,12 @@ func (a *clientFactoryAdapter) ClientFor(org, repo string) (RepoClient, error) {
 	return &repoClientAdapter{Repo: r}, err
 }
 
+// v1 clients do not support customizing pre-repo options.
```

Comment:
I did try this already, but then we have to delete one of our tests: https://github.com/kubernetes/test-infra/blob/1e81b5a526e375fd8953aecedb7d370e48a0f7a0/prow/config/inrepoconfig_test.go#L35. Doing it this way still lets that test pass (and I didn't think too deeply here but it felt cleaner to do this than to delete that test altogether).

Reply:
I think we need to remove that test then, right? If that causes the test to fail that is because this is a breaking change and we don't support the prow YAML getter with the v1 client anymore.

So I think we either need to always return an error from the unsupported function and remove the obsolete test, or properly implement this in the v1 client.

Reply:
I mean, technically the v1 client is supported for the prow YAML getter. But it will be less optimized than the v2 client, because the v1 client won't use the --sparse or --shared flags.

Granted, we are going to try to disable repo serialization altogether, so when we cross that bridge (soon), the v1 client will be basically unusable for the prow YAML getter. So I am OK with removing the obsolete test. Will update (err, this PR is merged now, so will have to do it as a follow-up).

Reply:
That followup is part of https://github.com/kubernetes/test-infra/pull/30428. It's kind of blocked on https://github.com/kubernetes/test-infra/pull/30520 though.